retrieval/discovery/ec2.go

```diff
@@ -85,6 +85,7 @@ func NewEC2Discovery(conf *config.EC2SDConfig) *EC2Discovery {
 		aws: &aws.Config{
 			Region:      &conf.Region,
 			Credentials: creds,
+			HTTPClient:  defaults.DefaultConfig.HTTPClient,
```

Comment:
I'm confused as to why we have to explicitly set a default, shouldn't it default to it? Also bear in mind that as I described above even though it may work with an environment variable due to the `http.DefaultClient`, that we are giving no stability guarantee on it.